src/fidgets/farcaster/Frame.tsx

```diff
@@ -0,0 +1,260 @@
+import {
+  FarcasterFrameContext,
+  FrameActionBodyPayload,
+  FrameUI,
+  fallbackFrameContext,
+} from "@frames.js/render";
+import type { FrameButton } from "frames.js";
+import { useFrame } from "@frames.js/render/use-frame";
+import Image from "next/image";
+import type { ImgHTMLAttributes } from "react";
+import { findIndex } from "lodash";
+import { useAuthenticatorManager } from "@/authenticators/AuthenticatorManager";
+import { createFarcasterSignerFromAuthenticatorManager } from ".";
+import {
+  CastId,
+  FarcasterNetwork,
+  FrameActionBody,
+  FrameActionMessage,
+  Message,
+  Signer,
+  makeFrameAction,
+} from "@farcaster/core";
+import { hexToBytes } from "@noble/ciphers/utils";
+
+// Due to issue with FrameImageNext from @frame.js/render/next
+// Implement the exact same thing again
+function FrameImageNext(
+  props: ImgHTMLAttributes<HTMLImageElement> & { src: string },
+): React.ReactNode {
+  return (
+    <Image
+      {...props}
+      alt={props.alt ?? ""}
+      sizes="100vw"
+      height={0}
+      width={0}
+    />
+  );
+}
+
+const FARCASTER_AUTHENTICATOR_NAME = "farcaster:nounspace";
+
+export type FrameFidgetSettings = {
+  url: string;
+};
+
+const frameConfig: FidgetEditConfig = {
+  fidgetName: "Frame",
+  fields: [
+    {
+      fieldName: "url",
+      required: true,
+      inputSelector: TextInput,
+    },
+  ],
+  size: {
+    minHeight: 1,
+    maxHeight: 36,
+    minWidth: 1,
+    maxWidth: 36,
+  },
+};
+
+async function createFrameActionMessage(
+  signer: Signer,
+  {
+    fid,
+    url,
+    buttonIndex,
+    castId,
+    inputText,
+    state,
+    address,
+    transactionId,
+  }: {
+    fid: number;
+    url: Uint8Array;
+    buttonIndex: number;
+    inputText: Uint8Array | undefined;
+    castId: CastId;
+    state: Uint8Array | undefined;
+    address: Uint8Array | undefined;
+    transactionId: Uint8Array | undefined;
+  },
+): Promise<
+  | {
+      message: null;
+      trustedBytes: null;
+    }
+  | {
+      message: FrameActionMessage | null;
+      trustedBytes: string;
+    }
+> {
+  const messageDataOptions = {
+    fid,
+    network: FarcasterNetwork.MAINNET,
+  };
+
+  const message = await makeFrameAction(
+    FrameActionBody.create({
+      url,
+      buttonIndex,
+      castId,
+      state,
+      inputText: inputText !== undefined ? Buffer.from(inputText) : undefined,
+      address,
+      transactionId,
+    }),
+    messageDataOptions,
+    signer,
+  );
+
+  if (message.isErr()) {
+    return { message: null, trustedBytes: null };
+  }
+
+  const trustedBytes = Buffer.from(
+    Message.encode(message._unsafeUnwrap()).finish(),
+  ).toString("hex");
+
+  return { message: message.unwrapOr(null), trustedBytes };
+}
+
+const Frame: React.FC<FidgetArgs<FrameFidgetSettings>> = async ({
+  settings: { url },
+}) => {
+  const authenticatorManager = useAuthenticatorManager();
+  const isLoadingSigner = await useMemo(
```

Comment:
I don't think useMemo is supposed to be used with async functions/promises, but not 100% sure.

Should use useEffect+useState instead, or a useAsyncMemo hook (something like https://github.com/awmleer/use-async-memo/blob/master/src/index.ts)

Reply:
Nevermind, I see you already fixed this in 179682ca46c7df875704a964c94da75d0f917493 👍